clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/cluster/AzureClusterProvider.groovy

```diff
@@ -0,0 +1,233 @@
+    cluster ? translateClusters([cluster], true)[0] : null
+  }
+
+  @Override
+  ServerGroup getServerGroup(String account, String region, String name) {
+    null
+  }
+
+  private Collection<AzureCluster> translateClusters(Collection<CacheData> clusterData, boolean includeDetails) {
+    Map<String, AzureLoadBalancer> loadBalancers
+    Map<String, AzureServerGroupDescription> serverGroups
+
+    if (includeDetails) {
+      Collection<CacheData> allLoadBalancers = resolveRelationshipDataForCollection(clusterData, AZURE_LOAD_BALANCERS.ns)
+      Collection<CacheData> allServerGroups = resolveRelationshipDataForCollection(clusterData, AZURE_SERVER_GROUPS.ns, RelationshipCacheFilter.include(AZURE_INSTANCES.ns))
+
+      loadBalancers = translateLoadBalancers(allLoadBalancers)
+      serverGroups = translateServerGroups(allServerGroups)
+    }
+
+    Collection<AzureCluster> clusters = clusterData.collect { CacheData clusterDataEntry ->
+      Map<String, String> clusterKey = Keys.parse(azureCloudProvider, clusterDataEntry.id)
+
+      def cluster = new AzureCluster()
+      cluster.accountName = clusterKey.account
+      cluster.name = clusterKey.name
+      cluster.type = azureCloudProvider.id
+      if (includeDetails) {
+        cluster.loadBalancers = clusterDataEntry.relationships[AZURE_LOAD_BALANCERS.ns]?.findResults {
+          loadBalancers.get(it)
+        }
+        cluster.serverGroups = clusterDataEntry.relationships[AZURE_SERVER_GROUPS.ns]?.findResults {
+          serverGroups.get(it)
+        }
+      } else {
+        cluster.loadBalancers = clusterDataEntry.relationships[AZURE_LOAD_BALANCERS.ns]?.collect { loadBalancerKey ->
+          Map parts = Keys.parse(azureCloudProvider, loadBalancerKey)
+          new AzureLoadBalancer(name: parts.name, region: parts.region, account: parts.account)
+        }
+
+        cluster.serverGroups = clusterDataEntry.relationships[AZURE_SERVER_GROUPS.ns]?.collect { serverGroupKey ->
+          Map parts = Keys.parse(azureCloudProvider, serverGroupKey)
+          new AzureServerGroupDescription(name: parts.serverGroup, region: parts.region,
+            application: parts.application, appName: parts.application)
+        }
+      }
+      cluster
+    }
+
+    clusters
+  }
+
+  private Map<String, AzureLoadBalancer> translateLoadBalancers(Collection<CacheData> loadBalancerData) {
+    loadBalancerData.collectEntries { loadBalancerEntry ->
+      Map<String, String> lbKey = Keys.parse(azureCloudProvider, loadBalancerEntry.id)
+      [(loadBalancerEntry.id) : new AzureLoadBalancer(name: lbKey.name, account: lbKey.account, region: lbKey.region)]
+    }
+  }
+
+  private Map<String, AzureServerGroupDescription> translateServerGroups(Collection<CacheData> serverGroupData) {
+
+    Map<String, AzureServerGroupDescription> serverGroups = serverGroupData.collectEntries { serverGroupEntry ->
```

Comment:
Do you need to have the serverGroups variable?  You don't on the translateLoadBalancer method and it looks like the exact same pattern.